dev_tools/oneapi/convert_source.sh

```diff
@@ -0,0 +1,429 @@
+    str_end="${str//[^>)\]]}"
+    if [[ "${#str_start}" -eq "${#str_end}" ]]; then
+        echo "true"
+    else
+        echo "false"
+    fi
+}
+
+convert_syntax() {
+    local syntax_regex="([^<>]*)(<[^<>]*>)?<<<(.*)>>>(.*)"
+    local str="$1"
+    str=$(echo "${str}" | sed -E 's/ +/ /g')
+    # echo "str ${str}"
+    local temp=""
+    local num=0
+    local var=""
+    if [[ "${str}" =~ ${syntax_regex} ]]; then
+        content="${BASH_REMATCH[3]}"
+        # echo "// content ${content}"
+        IFS=',' read -ra par <<< "$content"
+        for variable in "${par[@]}"; do
+            if [ -n "${temp}" ]; then
+                temp="${temp},"
+            fi
+            temp="${temp}${variable}"
+            # echo "// temp ${temp}"
+            is_closed=$(check_closed "$temp")
+            if [[ "${is_closed}" = "true" ]]; then
+                # echo "// temp ${temp}"
+                num=$((num+1))
+                # the last stream is zero, use the queue from exec
+                if [[ "${num}" -eq 4 ]] && [[ "${temp// }" -eq 0 ]]; then
+                    num=$((num-1))
+                else
+                    if [ -n "${var}" ]; then
+                        var="${var},"
+                    fi
+                    var="${var}${temp}"
+                fi
+                temp=""
+            fi
+        done
+        # echo "// var ${var}"
+        # echo "var ${var}"
+        if [[ "${num}" -lt 2 ]]; then
+            var="Error"
+        else
+            if [[ ${num} -eq 2 ]]; then
+                var="${var}, 0"
+                num=$((num+1))
+            fi
+            if [[ ${num} -eq 3 ]]; then
+                # var="${var}, exec->get_queue()"
+                var="${var}, GET_QUEUE"
+                num=$((num+1))
+            fi
+        fi
+        local suffix=""
+        local function_name=$(echo "${BASH_REMATCH[1]}" | sed -E 's/(.*::)| //g')
+        # echo "// functioname ${function_name}"
+        MAP_FILE="map_list"
+        suffix=$(cat "${MAP_FILE}" | sed -nE "s/${function_name} -> ${function_name}(.*)/\1/p")
+        suffix_matches=$(echo "${suffix}" | wc -l)
+        if [[ "${suffix_matches}" -gt 1 ]]; then
+            echo "static_assert(false, \"Has ${suffix_matches} matches for ${function_name}\");"
+            suffix=$(echo "${suffix}" | head -1)
+        fi
+
+        if [[ "${suffix}" = *"${CONFIG_SELECTION_SUFFIX}"* ]]; then
+            var="${var}, exec"
+        fi
+        local result="${BASH_REMATCH[1]}${suffix}${BASH_REMATCH[2]}(${var})${BASH_REMATCH[4]}"
+        result=$(echo "${result}" | sed -E 's/\) *\(+/,/g;s/, *\)/\)/g')
+        if [[ "${suffix}" = "" ]]; then
+            # if the function does not exist in the file, comment it.
+            echo "//remove//${result}"
+        else
+            echo "${result}"
+        fi
+    fi
+}
+
+
+convert_regex_allowed() {
+    local str="$1"
+    str=$(echo "${str}" | sed -E 's/(\/|\(|\)|\.)/\\\1/g')
+    echo "$str"
+}
```

Comment:
This one is sort of hard, I would appreciate a comment. Why use `\\\1`? It
looks like you are looking for different patterns using `|`. Would something
like the following work?
``` sed -E 's:([/().]{1)):\\\1/g' ```